### PR TITLE
Fix various issues related to attributes

### PR DESCRIPTION
**Pull request by JTS22**
Fixes multiple issues from #97

### CODE REVIEW

File: dpsim-models/src/EMT/EMT_Ph3_NetworkInjection.cpp [hunk 1]
```diff
@@ -29,8 +29,8 @@ EMT::Ph3::NetworkInjection::NetworkInjection(String uid, String name, Logger::Le
 	for (auto subcomp: mSubComponents)
 		mSLog->info("- {}", subcomp->name());
 
-	mVoltageRef->setReference(mSubVoltageSource->mVoltageRef);
-	mSrcFreq->setReference(mSubVoltageSource->mSrcFreq);
+	mSubVoltageSource->mVoltageRef->setReference(mVoltageRef);
```
[the remark this answers is not in the document]
Comment (JTS22) [reply dpsim-models/src/EMT/EMT_Ph3_NetworkInjection.cpp:32]: For a relationship between just two attributes, both options do the same, they set the attributes to point to the same value. However, when there is a third attribute involved which is referenced by one of the other attributes (e.g. when a new attribute is used for imports in the InterfaceVillas with the reference chain (Imported Attribute)<-(NetworkInjection::mVoltageRef)<-(mSubVoltageSource::mVoltageRef)), the order matters. With the previous code, configuring NetworkInjection::mVoltageRef to reference anything else  (e. g. through `intf->importAttr(evs->mVoltageRef)` would override the reference on the sub-voltagesource which means that the imported values are not correctly passed to the sub-voltagesource.

I agree that a more thorough explanation of reference relationships should be included in the documentation.

File: dpsim-models/src/EMT/EMT_Ph3_VoltageSource.cpp [hunk 2]
```diff
@@ -13,7 +13,7 @@ using namespace CPS;
 
 EMT::Ph3::VoltageSource::VoltageSource(String uid, String name, Logger::Level logLevel)
 	: SimPowerComp<Real>(uid, name, logLevel),
-	mVoltageRef(Attribute<MatrixComp>::create("V_ref", mAttributes)), // rms-value, phase-to-phase
+	mVoltageRef(Attribute<MatrixComp>::createDynamic("V_ref", mAttributes)), // rms-value, phase-to-phase
```
[the remark this answers is not in the document]
Comment (JTS22) [reply dpsim-models/src/EMT/EMT_Ph3_VoltageSource.cpp:16]: Yes! Both types are explained in quite some detail in #159

File: dpsim-models/src/EMT/EMT_Ph3_VoltageSource.cpp [hunk 3]
```diff
@@ -46,7 +46,6 @@ void EMT::Ph3::VoltageSource::setParameters(MatrixComp voltageRef, Real freqStar
 	mSrcSig = srcSigFreqRamp;
 
 	**mVoltageRef = voltageRef;
-	mSrcFreq->setReference(mSrcSig->mFreq);
```
[the remark this answers is not in the document]
Comment (JTS22) [reply dpsim-models/src/EMT/EMT_Ph3_VoltageSource.cpp:49]: I do not really see how that would make any sense, given that the `FrequencyRampGenerator` interpolates between a start and end frequency. Force-setting the current frequency by setting a reference on `mSrcFreq` would just turn it into a normal `SineWaveGenerator`, wouldn't it? And keeping it the old way (having `mSrcFreq` reference `mFreq`) is also kinda sketchy to me, given that `mSrcFreq` is normally used as an input parameter to the voltage source and not an output to be read.

File: dpsim-models/src/EMT/EMT_Ph3_VoltageSource.cpp [hunk 4]
```diff
@@ -58,7 +57,6 @@ void EMT::Ph3::VoltageSource::setParameters(MatrixComp voltageRef, Real modulati
 	mSrcSig = srcSigFm;
 
 	**mVoltageRef = voltageRef;
-	mSrcFreq->setReference(mSrcSig->mFreq);
```
[the remark this answers is not in the document]
Comment (JTS22) [reply dpsim-models/src/EMT/EMT_Ph3_VoltageSource.cpp:61]: As above, force-setting the frequency would defeat the purpose of the generator.

File: dpsim-models/include/dpsim-models/Components.h [hunk 5]
```diff
@@ -76,7 +76,6 @@
 #ifdef WITH_SUNDIALS
   #include <dpsim-models/DP/DP_Ph3_SynchronGeneratorDQODE.h>
 #endif
-#include <dpsim-models/DP/DP_Ph3_SynchronGeneratorVBR.h>
```
[the remark this answers is not in the document]
Comment (JTS22) [reply dpsim-models/include/dpsim-models/Components.h:79]: All the classes deleted in c1226b4622791af953fcc68db50db2a89afd5eb3 were malformed in the sense that they could not be compiled. The reason for this is that the .cpp files reference a class `SynchronGeneratorBase` which does not exist. Some .cpp files also include header-files that do not exist. None of these classes were included in the respective`CMakeLists.txt`, so they were never used in any (compiled) class or example.

I cannot comment on whether or not the models themselves are useful / meaningful. If this is the case, their code could be fixed and added back in. As long as this is not done though I would refrain from keeping them in the repo to not have any non-active, non-compilable code around, or at least clearly mark the files as incomplete / work-in-progress.

File: dpsim-models/include/dpsim-models/SP/SP_Ph1_PiLine.h [hunk 6]
```diff
@@ -36,23 +36,12 @@ namespace Ph1 {
 		// #### Power flow results ####
 		/// branch Current flow [A]
 		const Attribute<MatrixComp>::Ptr mCurrent;
-		/// CHECK: Are these derived attributes necessary?
-		Attribute<CPS::Complex>::Ptr mCurrent_0;
-		Attribute<CPS::Complex>::Ptr mCurrent_1;
 		
 		/// branch active powerflow [W], coef(0) has data from node 0, coef(1) from node 1.
```
[the remark this answers is not in the document]
Comment (JTS22) [reply dpsim-models/include/dpsim-models/SP/SP_Ph1_PiLine.h:40]: Done in 9649f20dab2efa3c8e7d4f3686cb730d177f6c28

File: dpsim-models/include/dpsim-models/Components.h (hunk 5 above)
[the remark this answers is not in the document]
Comment (dinkelbachjan) [reply dpsim-models/include/dpsim-models/Components.h:79]: The files belonging to `DP_Ph3_SynchronGeneratorVBR` could serve as basis for future implementation of the same. Though the class is not compilable, its files are restored in [e701270](https://github.com/sogno-platform/dpsim/pull/140/commits/e701270316e75b26d5d89f288a7f3a26dd26393f) and clearly marked as deprecated.

**Pull request merged**